sway/container.c

```diff
@@ -360,6 +392,38 @@ swayc_t *find_container(swayc_t *container, bool (*test)(swayc_t *view, void *da
 	return NULL;
 }
 
+swayc_t *swayc_by_handle(wlc_handle handle) {
+	return _swayc_by_handle_helper(handle, &root_container);
+}
+
+swayc_t *swayc_active_output(void) {
+	return root_container.focused;
+}
+
+swayc_t *swayc_active_workspace(void) {
+	return root_container.focused ? root_container.focused->focused : NULL;
+}
+
+swayc_t *swayc_active_workspace_for(swayc_t *cont) {
+	if (! cont) {
+		return NULL;
+	}
+	switch (cont->type) {
+	case C_ROOT: cont = cont->focused;
+	case C_OUTPUT: cont = cont->focused;
+	case C_WORKSPACE: return cont;
+	default: return swayc_parent_by_type(cont, C_WORKSPACE);
```

Comment:
style: split these up into more lines and add `/* fallthrough */` as appropriate